File: frontend/src/types/restaurant.ts
export interface Restaurant {
  id: string;
  name: string;
  address: string;
  lat: number;
  lng: number;
  rating?: number | null;
  price_level?: number | null;
  cuisine?: string | null;
  source: string;
  url?: string | null;
  num_reviews?: number | null;
}

File: backend/app/models/restaurant.py
from typing import Optional

from pydantic import BaseModel


class Restaurant(BaseModel):
    id: str
    name: str
    address: str
    lat: float
    lng: float
    rating: Optional[float] = None
    price_level: Optional[int] = None  # e.g. 1–4
    cuisine: Optional[str] = None
    source: str = "mock"  # 'google', 'yelp', etc.
    url: Optional[str] = None  # link to map / details
    num_reviews: Optional[int] = None
